p2p/validation/MESSAGE_VALIDATION_CONSENSUS.MD

```diff
@@ -0,0 +1,455 @@
+# Message Validation on Consensus messages
+
+## Summary
+
+This is a specification for the networking layer's message validation done on *QBFT* messages and *Decided* messages (aggregation of a quorum of commit messages). It is done by utilizing LibP2P's [extended validators](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md#extended-validators).
+The main goal of this validation is to prevent DOS attacks. In order to achieve this goal we apply some synchronicity assumptions to our communication model. We can do this because QBFT is partially synchronous, and Ethereum applies synchronicity assumptions as well.
+
+Every time a message passes our validation we create a mark for it.
+A mark saves data regarding the height, round and time of the message. The markers help us determine the validity of the next messages we receive.
+
+
+## Rationale and Design Goals
+
+The goal is to protect from malicious peers. In order to do so we want honest peers to not propagate messages that can be considered harmful, and disconnect from peers that seem harmful. The focus is on DOS attack that exhaust resources via redundant BLS signature verification.
+
+*Let's classify two type of attackers:*
+
+#### Non-signer attacker
+
+The attacker doesn't hold the the validator's partial key for which he pretains to send messages on behalf of. The attacker may hold partial keys of other validators but he is disincentivized from using them for attack purposes. This is because attacking operators in his quorum set may hinder or stop consensus, resulting in loss of potential operator's rewards.
+
+#### Signer attacker
+
+The attacker signs QBFT messages with a valid key. The loss of rewards is not meaningful to such attacker.
+
+*In addition let's classify two types of attack strategies:*
+
+#### Direct Attacker
+
+The attacker attempts to directly DOS nodes that it is peered to. Usually by bombarding the peers with messages that should exhaust the targets' CPU and memory. This attacker may be either aware or not aware of the implemented DOS protection measures.
+
+#### Covert Attacker
+
+This attacker attempts to DOS nodes in the network that it is not directly peered to. This is by spamming a different set of messages to its peers. The attacker is aware of the DOS protection and it chooses to spam its peers with messages that won't be picked up as malicious. However as those seemingly non-malicious messages propagate in the network, eventually some nodes will see a union of these messages that will trigger DOS protection measures. However, the attacker may avoid punishment if affected nodes won't be able to trace the source of the attack. Worse than that, affected nodes may punish their direct honest peers.
+
+
+## Specifications
+
+### Message Types
+
+1. *Decided* - An aggregated commit message that represents a quorum.
+2. *QBFT* - Propose, prepare, commit, and round-change messages.
```

Comment:
we should add partial sig messages as well

Reply:
TBD

Reply:
please mark it somewhere